ckanext/saml2/plugin.py

```diff
@@ -353,16 +389,22 @@ def _create_or_update_user(self, user_name, saml_info):
         context = {'schema': user_schema, 'ignore_auth': True}
 
         if is_new_user:
+            user_name = _get_random_username_from_email(saml_info['email'][0])
+            gen = saml_info['id'][0]
+            data_dict['id'] = unicode(uuid.uuid4())
+            data_dict['name'] = user_name
             log.debug("Creating user: %s", data_dict)
             data_dict['password'] = self.make_password()
-            p.toolkit.get_action('user_create')(context, data_dict)
+            new_user = p.toolkit.get_action('user_create')(context, data_dict)
             assign_default_role(context, user_name)
-
+            model.Session.add(UserSsoGen(id=new_user['id'],
+                                         gen=gen,
+                                         user_name=user_name))
+            model.Session.commit()
         elif update_user:
             log.debug("Updating user: %s", data_dict)
             p.toolkit.get_action('user_update')(context, data_dict)
-
-        return model.User.get(user_name)
+        return model.User.get(user_name) if user_name is not None else []
```

Comment:
Are you sure, that you want to return empty list instead of either None or `User` object?